cmd/txtar-c/savedir.go

```diff
@@ -105,7 +105,13 @@ func main1() int {
 		})
 		return nil
 	})
-
+	if *scriptFlag != "" {
+		script, err := os.ReadFile(*scriptFlag)
+		if err != nil {
+			log.Fatal(err)
+		}
+		a.Comment = append(a.Comment, script...)
```

Comment:
What about the scenario when the `-script` argument contains unquote lines itself?

Reply:
I'm not sure if I see what you mean. Could you say a little more?

Reply:
He means that if the script file contains txtar syntax, like `-- foo --`, then it will leak out of the comment section and affect the rest of the txtar.

The flag should probably reject those scripts, because I don't think there is any easy quoting or escaping we can reach for.

Reply:
Sorry, I still don't get it. Could you give an example of a script which would produce the wrong result?

Reply:
I think I actually misunderstood what Paul said, so ignore me.

Reply:
> I'm not sure I understand the question. If you mean "only the comment part of the script makes it into the resulting `txtar`, ignoring any file inclusions", that's not the case. The whole of the file referenced by `-script FILE` is included, and if it contains file inclusions, they will end up in the final `txtar` too. At least, that was the intention.

I actually misread the code. As you say, the entire file passed to `-script` is included after `unquote` lines that are added by `txtar-c`. What threw me is that the contents of this file are added to the new archive's comments, even though that `[]byte` contains files as interpreted by `txtar`. 

Here's a case where I think things go wrong:

```
unquote expect
unquote test.txtar
unquote blah/needsquote.txtar
# With both -script and -quote, it should include test.txtar after any 'unquote's
exec txtar-c -quote -script test.txtar blah
cmp stdout expect

-- test.txtar --
>unquote needsquote.txtar
>exec cat my.txtar
>
>-- needsquote.txtar --
>># This is a comment in my.txtar
>>
>>-- a.txt --
>>// This is a.txt within my.txtar
>>
-- blah/go.mod --
module example.com/blah

-- blah/main.go --
package main

import "fmt"

func main() {
  fmt.Println("Hello, world!")
}

-- blah/needsquote.txtar --
>-- file_entry.txt --

-- expect --
>unquote needsquote.txtar
>unquote needsquote.txtar
>exec cat my.txtar
>
>-- needsquote.txtar --
>># This is a comment in my.txtar
>>
>>-- a.txt --
>>// This is a.txt within my.txtar
>>
>-- go.mod --
>module example.com/blah
>
>-- main.go --
>package main
>
>import "fmt"
>
>func main() {
>  fmt.Println("Hello, world!")
>}
>
>-- needsquote.txtar --
>>-- file_entry.txt --
>>
```

The double `unquote` of `needsquote.txtar` is incorrect; there should only be one `unquote`. 

The flag name `-script` is also a bit of a smell for me here. `txtar-c` is a command that is only concerned with the `txtar` format. It knows nothing about `testscript`. If we take the name "script" away, we are effectively passing in another `txtar` archive as the argument here. In that case, how would you describe the operation we are performing on what are two `txtar` archives? Merge? Append?

> If you mean "only the comment part of the resulting `txtar` is modified, by including the script in it", that's correct, but I can't think of an alternative way of doing this that would make sense.
> 
> > my question is really directed at questioning whether this could be adequately solved in a purpose built program, rather than trying to adapt txtar-c.
> 
> Certainly it could.

My sense is that we're struggling to define what this operation between two `txtar` archives is here. And as I said, I _think_ that comes down to the fact that in the general case, operations on archives are not well defined: they don't compose, prepend, append etc. Hence I feel this probably belongs as a purpose-built tool for your situation, rather than something that is generally useful in `txtar-c`.

Reply:
>  I feel this probably belongs as a purpose-built tool for your situation, rather than something that is generally useful in txtar-c.

That's perfectly fair—and I _have_ exactly that tool, which is working perfectly for my use case. (It is, in fact, just vanilla `txtar-c` with this patch applied.)

I feel the sense of the meeting is that while a feature like this might be occasionally useful, to provide a really robust solution that works in _theory_, as well as merely in practice, would take a fair bit more work, if it's even possible at all. Accordingly, please close, with my thanks.